### PR TITLE
done. check it

### PR DESCRIPTION
**Pull request by Watxesnavo**

### CODE REVIEW

File: src/main/java/mate/academy/model/Order.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+@Entity
+@Table(name = "orders")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
+    private LocalDateTime orderDate;
```
Comment (olekskov) [inline src/main/java/mate/academy/model/Order.java:21]: Consider adding a @Column(name = "order_date") annotation to map the orderDate field to the 'order_date' column in the 'orders' table.

File: src/main/java/mate/academy/Main.java [hunk 2]
```diff
@@ -36,20 +47,37 @@ public static void main(String[] args) {
...
 
         MovieSession yesterdayMovieSession = new MovieSession();
         yesterdayMovieSession.setCinemaHall(firstCinemaHall);
         yesterdayMovieSession.setMovie(fastAndFurious);
+        yesterdayMovieSession.setMovie(dumbEslava);
         yesterdayMovieSession.setShowTime(LocalDateTime.now().minusDays(1L));
 
-        MovieSessionService movieSessionService = null;
+        MovieSessionService movieSessionService =
+                (MovieSessionService) injector.getInstance(MovieSessionService.class);;
```
Comment (olekskov) [inline src/main/java/mate/academy/Main.java:61]: Remove the extra semicolon at the end of the line.

File: src/main/java/mate/academy/Main.java [hunk 3]
```diff
@@ -27,7 +37,8 @@ public static void main(String[] args) {
         secondCinemaHall.setCapacity(200);
         secondCinemaHall.setDescription("second hall with capacity 200");
 
-        CinemaHallService cinemaHallService = null;
+        CinemaHallService cinemaHallService =
+                (CinemaHallService) injector.getInstance(CinemaHallService.class);;
```
Comment (olekskov) [inline src/main/java/mate/academy/Main.java:41]: Remove the extra semicolon at the end of the line.

File: src/main/java/mate/academy/dao/impl/OrderDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    public Order getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Order> query = session.createQuery("FROM Order ord "
+                    + "LEFT JOIN FETCH ord.tickets t "
+                    + "LEFT JOIN FETCH t.movieSession ms "
+                    + "LEFT JOIN FETCH ms.movie "
+                    + "LEFT JOIN FETCH ms.cinemaHall "
+                    + "WHERE ord.user =:user", Order.class);
+            query.setParameter("user", user);
+            return query.uniqueResult();
```
Comment (olekskov) [inline src/main/java/mate/academy/dao/impl/OrderDaoImpl.java:47]: Replace 'query.uniqueResult()' with 'query.getResultList()' to fetch all orders for the given user.

File: src/main/java/mate/academy/dao/impl/OrderDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            throw new DataProcessingException("Can't insert a order: " + order, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Order getByUser(User user) {
```
Comment (olekskov) [inline src/main/java/mate/academy/dao/impl/OrderDaoImpl.java:38]: According to the task, the method should return 'order history for a user', but the current implementation returns only one order. You should return a List of Order objects instead.

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        order.setOrderDate(LocalDateTime.now());
+        orderDao.add(order);
+        shoppingCartService.clearShoppingCart(shoppingCart);
+        return order;
+    }
+
+    @Override
+    public List<Order> getOrdersHistory(User user) {
+        List<Order> orders = new ArrayList<>();
+        orders.add(orderDao.getByUser(user));
```
Comment (olekskov) [inline src/main/java/mate/academy/service/impl/OrderServiceImpl.java:36]: The method orderDao.getByUser should return a list of orders. Change the method signature to return a list and update the implementation accordingly.

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        for (Order o : orders) {
+            if (!o.getId().equals(orderDao.getByUser(user).getId())) {
```
Comment (olekskov) [inline src/main/java/mate/academy/service/impl/OrderServiceImpl.java:38]: The loop and the if condition are unnecessary. Remove them and directly return the result from orderDao.getByUser.

File: src/main/java/mate/academy/dao/OrderDao.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package mate.academy.dao;
+
+import mate.academy.model.Order;
+import mate.academy.model.User;
+
+public interface OrderDao {
+    Order add(Order order);
+
+    Order getByUser(User user);
```
Comment (olekskov) [inline src/main/java/mate/academy/dao/OrderDao.java:9]: According to task it should be:
```suggestion
    List<Order> getByUser(User user);
```

File: src/main/java/mate/academy/model/Order.java [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "orders")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
```
Comment (olekskov) [inline src/main/java/mate/academy/model/Order.java:20]: it's better to add joinTable annotation and set table and columns names there

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/dao/impl/OrderDaoImpl.java (hunk 4 above)
Comment (Watxesnavo) [reply src/main/java/mate/academy/dao/impl/OrderDaoImpl.java:47]: method doesn't return a list of objects to put resultlist... it returns one concrete object, its like the readme file has

File: src/main/java/mate/academy/dao/impl/OrderDaoImpl.java (hunk 5 above)
Comment (Watxesnavo) [reply src/main/java/mate/academy/dao/impl/OrderDaoImpl.java:38]: public interface OrderService {
      Order completeOrder(ShoppingCart shoppingCart);
      
      List<Order> getOrdersHistory(User user);
  }
that is what it looks like in readme file

**Review by Rommelua — APPROVED** [review]

**Pull request closed without merging**